### PR TITLE
chore: Update containerd config version 2

### LINKED ISSUE #3816: Update containerd config file version
## Feature Request

https://containerd.io/releases/#deprecated-features

And disabled_plugins = ["aufs", "zfs", "devmapper"], Talos does not have it.

### PR DESCRIPTION
**Pull request by sergelogvinov**
Fixes #3816

* Rename key cri -> io.containerd.grpc.v1.cri
* Disable plugins aufs,zfs,devmapper,btrfs (less warning messages on
  boot time)

### CODE REVIEW

File: hack/containerd.toml [hunk 1]
```diff
@@ -1,4 +1,8 @@
+version = 2
+
+disabled_plugins = ["io.containerd.snapshotter.v1.aufs", "io.containerd.v1.zfs", "io.containerd.snapshotter.v1.zfs", "io.containerd.v1.devmapper", "io.containerd.snapshotter.v1.devmapper", "io.containerd.snapshotter.v1.btrfs"]
```
Comment (andrewrynhard) [inline hack/containerd.toml:3]: 👍🏻

**Comment by smira** [comment]
pushing this for 0.12 (probably next week we can start landing PRs)

**Comment by smira** [comment]
I feel it might need a bit of work to eventually catch up with `containerd` config:

```
172.20.0.2: time="2021-06-30T20:15:55.769537354Z" level=warning msg="`mirrors` is deprecated, please use `config_path` instead"
```

but this doesn't really depend on the containerd config format

**Review by smira — DISMISSED** [review]

**Comment by smira** [comment]
unit-test failure:

```
#166 150.4 === RUN   TestConfigSuite/TestGenerateRegistriesConfig
#166 150.4     config_test.go:79: 
#166 150.4         	Error Trace:	config_test.go:79
#166 150.4         	Error:      	Not equal: 
#166 150.4         	            	expected: []config.File{(*v1alpha1.MachineFile)(0xc000b1fa80), (*v1alpha1.MachineFile)(0xc000b1fac0), (*v1alpha1.MachineFile)(0xc000b1fb00), (*v1alpha1.MachineFile)(0xc000b1fb40)}
#166 150.4         	            	actual  : []config.File{(*v1alpha1.MachineFile)(0xc000b1f5c0), (*v1alpha1.MachineFile)(0xc000b1f600), (*v1alpha1.MachineFile)(0xc000b1f640), (*v1alpha1.MachineFile)(0xc000b1fa40)}
#166 150.4         	            	
#166 150.4         	            	Diff:
#166 150.4         	            	--- Expected
#166 150.4         	            	+++ Actual
#166 150.4         	            	@@ -20,3 +20,3 @@
#166 150.4         	            	  (*v1alpha1.MachineFile)({
#166 150.4         	            	-  FileContent: (string) (len=791) "[plugins]\n  [plugins.cri]\n    [plugins.cri.registry]\n      [plugins.cri.registry.mirrors]\n        [plugins.cri.registry.mirrors.\"docker.io\"]\n          endpoint = [\"https://registry-1.docker.io\", \"https://registry-2.docker.io\"]\n      [plugins.cri.registry.configs]\n        [plugins.cri.registry.configs.\"some.host:123\"]\n          [plugins.cri.registry.configs.\"some.host:123\".auth]\n            username = \"root\"\n            password = \"secret\"\n            auth = \"auth\"\n            identitytoken = \"token\"\n          [plugins.cri.registry.configs.\"some.host:123\".tls]\n            insecure_skip_verify = true\n            ca_file = \"/var/etc/cri/ca/some.host:123.crt\"\n            cert_file = \"/var/etc/cri/client/some.host:123.crt\"\n            key_file = \"/var/etc/cri/client/some.host:123.key\"\n",
#166 150.4         	            	+  FileContent: (string) (len=983) "[plugins]\n  [plugins.\"io.containerd.grpc.v1.cri\"]\n    [plugins.\"io.containerd.grpc.v1.cri\".registry]\n      [plugins.\"io.containerd.grpc.v1.cri\".registry.mirrors]\n        [plugins.\"io.containerd.grpc.v1.cri\".registry.mirrors.\"docker.io\"]\n          endpoint = [\"https://registry-1.docker.io\", \"https://registry-2.docker.io\"]\n      [plugins.\"io.containerd.grpc.v1.cri\".registry.configs]\n        [plugins.\"io.containerd.grpc.v1.cri\".registry.configs.\"some.host:123\"]\n          [plugins.\"io.containerd.grpc.v1.cri\".registry.configs.\"some.host:123\".auth]\n            username = \"root\"\n            password = \"secret\"\n            auth = \"auth\"\n            identitytoken = \"token\"\n          [plugins.\"io.containerd.grpc.v1.cri\".registry.configs.\"some.host:123\".tls]\n            insecure_skip_verify = true\n            ca_file = \"/var/etc/cri/ca/some.host:123.crt\"\n            cert_file = \"/var/etc/cri/client/some.host:123.crt\"\n            key_file = \"/var/etc/cri/client/some.host:123.key\"\n",
#166 150.4         	            	   FilePermissions: (v1alpha1.FileMode) 420,
#166 150.4         	Test:       	TestConfigSuite/TestGenerateRegistriesConfig
```

**Comment by sergelogvinov** [comment]
> I feel it might need a bit of work to eventually catch up with `containerd` config:
> 
> ```
> 172.20.0.2: time="2021-06-30T20:15:55.769537354Z" level=warning msg="`mirrors` is deprecated, please use `config_path` instead"
> ```
> 
> but this doesn't really depend on the containerd config format

I think, we will fix it here #3836

**Review by smira — APPROVED** [review]

**Pull request merged**